### PR TITLE
bu: resolve ccaching of artifacts #1826

### PR DESCRIPTION
**Pull request by mlyle**
Needs test of build on Windows.

### CODE REVIEW

**Comment by BrainFPV** [comment]
Nice workaround :). LGTM.

File: flight/targets/bu/f4/Makefile [hunk 1]
```diff
@@ -81,12 +81,18 @@ EXTRAINCDIRS  += $(BUBOARDDIR)
...
 CDEFS += -DMEM_SIZE=$(FW_BANK_SIZE)
 CDEFS += -DUSE_$(BOARD)
-CDEFS += -DBU_PAYLOAD_FILE=$(ROOT_DIR)/build/bl_$(BOARD_NAME)/bl_$(BOARD_NAME).bin
+CDEFS += -DBU_PAYLOAD_FILE="$(PAYLOAD_FILE)"
 
+# This exists to prevent ccache from caching compilation results when the file
+# has changed.  md5sum is used on linux and Windows; cksum is used on mac.
+CDEFS += -DBU_HASH=$(shell md5sum "$(PAYLOAD_FILE)" 2>/dev/null || cksum "$(PAYLOAD_FILE)" | cut -d ' ' -f 1)
+
+$(shell md $(subst /,\\,$(OUTDIR)) 2>NUL)
```
Comment (peabody124) [inline flight/targets/bu/f4/Makefile]: is this line not needed on all targets?

Comment (mlyle) [inline flight/targets/bu/f4/Makefile]: Woops, accidentally duplicated from below.  Removing.

**Comment by mlyle** [comment]
This is interesting because it doesn't happen on my linux box or mac.

**Comment by mlyle** [comment]
OK fixed.  Need to verify build on windows before commitin' this one.

**Comment by mlyle** [comment]
Need just a little more vetting.  Sometimes it's the simplest things that are hardest.

**Comment by mlyle** [comment]
It's all good.

**Comment by jihlein** [comment]
This breaks the bu build process on my windows 7 build machine.  Reverting to the previous makefile version fixes it.

The error is as follows:

$ make bu_coptercontrol
mingw32-make[1]: Nothing to be done for 'bin'.
 AS          flight/PiOS/STM32F10x/startup_stm32f10x_MD_CC.S
 CC          flight/targets/bu/common/main.c
In file included from c:/TauLabs/flight/PiOS/pios.h:61:0,
                 from c:/TauLabs/flight/targets/bu/common/main.c:32:
c:/TauLabs/flight/PiOS/STM32F10x/Libraries/CMSIS/Core/CM3/stm32f10x.h:478:22: fatal error: core_cm3.h: No such file or directory
 #include "core_cm3.h"
                      ^
compilation terminated.
makefile:194: recipe for target 'c:/TauLabs/build/bu_coptercontrol/main.o' failed
mingw32-make[1]: *** [c:/TauLabs/build/bu_coptercontrol/main.o] Error 1
makefile:891: recipe for target 'bu_coptercontrol_tlfw' failed
mingw32-make: *** [bu_coptercontrol_tlfw] Error 2

**Comment by mlyle** [comment]
Thanks for the find.. fixing up.

**Pull request merged**